chain/index/api.go

```diff
@@ -126,17 +127,124 @@ func (si *SqliteIndexer) ChainValidateIndex(ctx context.Context, epoch abi.Chain
 		return nil, xerrors.Errorf("failed to cast tipset key cid: %w", err)
 	}
 
-	if indexedTsKeyCid != expectedTsKeyCid {
+	if !indexedTsKeyCid.Equals(expectedTsKeyCid) {
 		return nil, xerrors.Errorf("index corruption: non-reverted tipset at height %d has key %s, but canonical chain has %s", epoch, indexedTsKeyCid, expectedTsKeyCid)
 	}
 
+	// indexedTsKeyCid and expectedTsKeyCid are the same, so we can use `expectedTs` to fetch the indexed data
+	indexedData, err := si.getIndexedTipSetData(ctx, expectedTs.Key())
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get tipset message and event counts at height %d: %w", expectedTs.Height(), err)
+	}
+
+	if err = si.verifyIndexedData(ctx, expectedTs, indexedData); err != nil {
+		return nil, xerrors.Errorf("failed to verify indexed data at height %d: %w", expectedTs.Height(), err)
+	}
+
 	return &types.IndexValidation{
-		TipsetKey: expectedTs.Key().String(),
-		Height:    uint64(expectedTs.Height()),
-		// TODO Other fields
+		TipsetKey:      expectedTs.Key().String(),
+		Height:         uint64(expectedTs.Height()),
+		TotalMessages:  uint64(indexedData.nonRevertedMessageCount),
+		TotalEvents:    uint64(indexedData.nonRevertedEventCount),
+		EventsReverted: indexedData.hasRevertedEvents,
 	}, nil
 }
 
+type indexedTipSetData struct {
+	nonRevertedMessageCount int
+	nonRevertedEventCount   int
+	hasRevertedEvents       bool
```

Comment:
Can you add a test for these fields to `TestEthGetLogsBasic` in `eth_filtest_test.go` ?

We already call this API there and so this should be easy. See 

```
iv, err := client.ChainValidateIndex(ctx, abi.ChainEpoch(0), false)
	require.NoError(err)
	require.NotNil(iv)

	fmt.Printf("index validation: %v\n", iv)

	iv, err = client.ChainValidateIndex(ctx, abi.ChainEpoch(22), false)
	require.NoError(err)
	require.NotNil(iv)
	fmt.Printf("index validation: %v\n", iv)
```

Reply:
The test is failing right now because `loadExecutedMessage` is not loading any messages, and we are using it to verify the indexed data.